CMakeLists.txt

```diff
@@ -136,13 +136,26 @@ include(utils)
 find_package(hsa-runtime64 PATHS /opt/rocm )
 if(${hsa-runtime64_FOUND})
   message("hsa-runtime64 found @  ${hsa-runtime64_DIR} ")
+  set(RBT_HSA_VERSION_MAJ ${hsa-runtime64_VERSION_MAJOR} )
+  set(RBT_HSA_VERSION_MIN ${hsa-runtime64_VERSION_MINOR} )
+  set(RBT_HSA_VERSION_PATCH ${hsa-runtime64_VERSION_PATCH} )
+  #Rocm 5.3.0 will have hsa file reorg changes. The package version is 1055300
+  math(EXPR RBT_HSA_VERSION_FILEREORG "1056300")
+  add_compile_options(-DRBT_HSA_VERSION_FILEREORG=${RBT_HSA_VERSION_FILEREORG})
+
+  if( (RBT_HSA_VERSION_MAJ GREATER 999) OR (RBT_HSA_VERSION_MIN GREATER 999) )
+    message(FATAL_ERROR "RBT hsa-runtime64: Too big HSA runtime version number(s)" )
+  else() #if Valid hsa Version flags
+    math(EXPR RBT_HSA_VERSION_FLAT  "(${RBT_HSA_VERSION_MAJ}*1000000+${RBT_HSA_VERSION_MIN}*1000+${RBT_HSA_VERSION_PATCH})" )
+    add_compile_options(-DRBT_HSA_VERSION_FLAT=${RBT_HSA_VERSION_FLAT})
+  endif()
 else()
```

Comment:
Please add a comment that the else branch is for the case we are binding to the old rules of the header

Reply:
Here we are finding the package version and use that value in source file to decide whether old style "hsa.h"or new style "hsa/hsa.h" need to be used
if no hsa package found and if the version is invalid, then the default case will be "hsa/hsa.h" , since the old style will get deprecated